### ISSUE TITLE
Gains identified from normal walking are very similar to that of perturbed walking

### ISSUE THREAD

**Issue by moorepants** [csu-hmc/gait-control-direct-id-paper#41]
This plot shows mean joint isolated gains identified from normal walking across all trials at 1.2 m/s:
![mean-gains-1 2](https://cloud.githubusercontent.com/assets/276007/7508402/77151e78-f434-11e4-9a67-96865c82b143.png)
And this plots shows the same thing from perturbed walking:
![mean-gains-1 2](https://cloud.githubusercontent.com/assets/276007/7508405/87609ab4-f434-11e4-9f49-d896203cea0a.png)
The proportional gains (top row) have striking similarities. The ankle derivative gains are similar. The other two derivative gains are less similar.

On the optimistic side this means that you don't need to perturb people to identify their controller, i.e. they perturb themselves enough.

But this could also just show that our controller input and output measurements are correlated through the marker coordinates, see issue #40. If the correlation is dominant then maybe we are just seeing that in both results and this is meaningless.

Note that the difference in the standard deviation in the joint angle, rate, and torque measurements across gait cycles is pretty small. It isn't clear that the perturbed walking actually varies the motion that much. So maybe this is expected.

To prove that the person self-perturbs themselves enough maybe it is worth running the identification on perfectly cyclic walking. We should get that K = 0 in that case. If I then identify from a large number of gait cycles except starting with all of them the same, then half of them the same, then a quarter, etc, would these patterns start to appear? Or maybe scaling all the gait cycles towards the mean. If I could crank up and down the variation in the gait cycles. Find the mean gait cycle, subtract the mean from each gait cycle leaving the difference, then scale these differences.

@tvdbogert Any comments on this? I need to explain what this means.

**Comment by moorepants** [comment]
My idea about scaling the input and outputs to the identifier was a dumb idea. If you scale both the inputs and outputs then the same correlation is found. This can probably only be done with actually cranking up and down the perturbations. Doing it with a simulation would be ideal, but you need a controller that can handle perturbations first.

**Comment by moorepants** [comment]
I spent most of today talking with @monthubbard and Andy Ruina about this work. Both of them seem to think this is likely a good finding, i.e. that perturbations are not necessary to make the data rich enough for control identification. I essentially agree, but I want to feel comfortable that this is identifying a controller at all. If I can convince myself of that, then this can likely be stated as a discovery.

**Comment by tvdbogert** [comment]
I don't think this comes from correlations via marker coordinates (issue #40, which I still need to comment on).  Those proportional gains were lower, and the derivative gains were essentially zero.

I tend to me optimistic about this one also, and would suggest that people perturb themselves sufficiently to make control identification possible.  But of course there is concern that without sufficient perturbations we are only seeing the inverse of the plant.  van der Kooij proved this for a simple system, but this system is more complex.  We are not seeing all of the actuation.  We should just state this in the discussion as an open question.  We could decide to not even present the unperturbed results, but I feel that's not ethical.  If people can do this identification without any perturbation, that is a huge simplification of the protocol and equipment and important to know.

Another thing we can conclude from this: since both sets of results are the same (perturbed and not perturbed), they are either both correct or both wrong.

**Comment by moorepants** [comment]
> I don't think this comes from correlations via marker coordinates

Ok, the #40  is my main hiccup for moving forward and I'm still lost. I have a simple model that I'm working on that may help shed some light there though. That's my only path forward at the moment.

> I tend to me optimistic about this one also, and would suggest that people perturb themselves sufficiently to make control identification possible.

I'm leaning towards this now too, but not sure about justifying it scientifically. I think this may be true in general for human control id. We just have a noisy system that keeps us in a state of perturbation. These perturbations may be small, but if we can measure them, then we can id from it.

> But of course there is concern that without sufficient perturbations we are only seeing the inverse of the plant

It is very clear to me how you may identify the inverse of the plant when using frequency domain analyses. But in time domain id, if your controller structure isn't complex enough to mathematically represent the inverse of the plant, how can you identify the inverse of the plant? For example, our controller doesn't have terms proportional to the angular acceleration. In this case, does getting the "inverse of the plant" simply mean that we will get erroneous gain values. I feel like that is what you all concluded in Samin's thesis.

> We could decide to not even present the unperturbed results, but I feel that's not ethical.

I feel like I have to show this. I'd feel unethical if I didn't.

> Another thing we can conclude from this

Yes I agree.

The only other information that I have is Samin's result for standing. She shows how the gain bias is affected by the magnitude of perturbation. I have here numbers and compared them here:

http://nbviewer.ipython.org/github/csu-hmc/gait-control-direct-id-paper/blob/master/notebooks/artificial_variations.ipynb#Compare-the-standard-deviation-in-the-measurements

I compared the standard deviation in joint angles/torques to see if they were more than she needed for id purposes in standing to reduce the gain bias.
